src/main/java/org/commcare/core/process/CommCareInstanceInitializer.java

```diff
@@ -89,10 +90,27 @@ public InstanceRoot generateRoot(ExternalDataInstance instance) {
             return setupRemoteData(instance);
         } else if (ref.contains("migration")) {
             return setupMigrationData(instance);
+        } else if (ref.contains("selected_cases")) {
+            return setupSelectedCases(instance);
         }
         return ConcreteInstanceRoot.NULL;
     }
 
+    protected InstanceRoot setupSelectedCases(ExternalDataInstance instance) {
```

Comment:
Would be good to add javadoc with a note that this should be implemented by subclasses.